.github/workflows/nightly.yml

```diff
@@ -0,0 +1,47 @@
+
+# This workflow includes more expensive tests than what is run on every PR, that are unlikely to break on most changes.
+# It also publishes a nightly release.
+name: Nightly test and release workflow
+
+on:
+  schedule:
+    # Chosen to be hopefully outside of business hours for most contributors'
+    # time zones, and not on the hour to avoid heavy scheduled-job times:
+    # https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#schedule
+    - cron: "30 6 * * *"
+  workflow_dispatch:
+
+jobs:
+  deep-integration-tests:
+    uses: ./.github/workflows/integration-tests-reusable.yml
+    with:
+      all_platforms: true
+      num_shards: 5
+
+  determine-vars:
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Get short sha
+        id: short-sha
+        run: echo "::set-output name=sha_short::$(git rev-parse --short HEAD)"
+
+      - name: Get current date
+        id: date
+        run: echo "::set-output name=date::$(date +'%Y-%m-%d')"
+
+    outputs:
+      short-sha: ${{ steps.short-sha.outputs.sha_short }}
+      date: ${{ steps.date.outputs.date }}
+      name: nightly-${{ steps.date.outputs.date }}-${{ steps.short-sha.outputs.sha_short }}
+
+  publish-release:
+    uses: ./.github/workflows/publish-release-reusable.yml
+    with:
+      sha: ${{ refs/heads/master }}
+      name: ${{ needs.determine-vars.outputs.name }}
+      tag_name: nightly
+      prerelease: true
+      release_nuget: false
```

Comment:
Fine for v1, but it would be really nice to publish these to nuget as well, so we can move towards that being the only way to consume Dafny. See https://docs.microsoft.com/en-us/nuget/create-packages/prerelease-packages.